crates/sui-core/src/consensus_manager/mysticeti_manager.rs

```diff
@@ -107,81 +102,69 @@ impl ConsensusManagerTrait for MysticetiManager {
-            // is the one we want to go with.
-            #[allow(clippy::disallowed_methods)]
-            let (commit_sender, commit_receiver) = unbounded_channel();
-
-            let consensus_handler = consensus_handler_initializer.new_consensus_handler();
-            let consumer = CommitConsumer::new(
-                commit_sender,
-                consensus_handler.last_executed_sub_dag_index().await,
-            );
-
-            match Validator::start_production(
-                authority_index,
-                committee.clone(),
-                &parameters,
-                config.clone(),
-                registry.clone(),
-                Signer(Box::new(private_key.0.clone())),
-                consumer,
-                tx_validator.clone(),
-            )
-            .await
-            {
-                Ok((validator, tx_sender)) => {
-                    let registry_id = self.registry_service.add(registry);
-
-                    self.validator
-                        .swap(Some(Arc::new((validator, registry_id))));
-
-                    // create the client to send transactions to Mysticeti and update it.
-                    self.client.set(MysticetiClient::new(tx_sender));
-
-                    // spin up the new mysticeti consensus handler to listen for committed sub dags
-                    let handler =
-                        MysticetiConsensusHandler::new(consensus_handler, commit_receiver);
-                    self.consensus_handler.store(Some(Arc::new(handler)));
-
-                    break;
-                }
-                Err(err) => {
-                    retries += 1;
-
-                    self.metrics.start_mysticeti_retries.set(retries as i64);
-
-                    if retries >= MAX_RETRIES {
-                        panic!(
-                            "Failed starting Mysticeti, maxed out retries {}: {:?}",
-                            retries, err
-                        );
-                    }
-
-                    tracing::error!("Failed starting Mysticeti, retry {}: {:?}", retries, err);
-                    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
-                }
-            }
-        }
+        // TODO: that should be replaced by a metered channel. We can discuss if unbounded approach
+        // is the one we want to go with.
+        #[allow(clippy::disallowed_methods)]
+        let (commit_sender, commit_receiver) = unbounded_channel();
+
+        let consensus_handler = consensus_handler_initializer.new_consensus_handler();
+        let consumer = CommitConsumer::new(
+            commit_sender,
+            // TODO(mysticeti): remove dependency on narwhal executor
+            consensus_handler.last_executed_sub_dag_index().await,
+        );
+
+        // TODO(mysticeti): Investigate if we need to return potential errors from
+        // AuthorityNode and add retries here?
```

Comment:
The main reason for adding retries is due to of transient errors having to do with some resource allocation from previous that didn't get released successfully yet. For instance we saw such examples during network shutdown where the port wasn't released yet while trying to spin up the new instance, but it rather needed a few millis before do so. I think having the retries is ok but we can discuss about it.